R/methods-KPartiteNetwork.R

```diff
@@ -1,4 +1,54 @@
-#' @include methods-Nodes.R
 ## Methods for Partitions
+#' @include class-KPartiteNetwork.R
+setGeneric("partitions", function(object) standardGeneric("partitions"))
+setGeneric("partitions<-", function(object, value) standardGeneric("partitions<-"))
+
+setMethod("partitions", "KPartiteNetwork", function(object) object@partitions)
+setMethod("partitions<-", "KPartiteNetwork", function(object, value) {object@partitions <- value; validObject(object); object})
+
+#' @include methods-Nodes.R
 setGeneric("getAllNodeIds", function(object) standardGeneric("getAllNodeIds"))
-setMethod("getAllNodeIds", "Partitions", function(object) unlist(lapply(as.list(object), getNodeIds)))
\ No newline at end of file
+setMethod("getAllNodeIds", "Partitions", function(object) unlist(lapply(as.list(object), getNodeIds)))
+
+toJSONGeneric <- getGeneric("toJSON", package = "veupathUtils")
+
+#' Convert Partitions object to JSON
+#' 
+#' Converts a Partitions object to JSON
+#' @param object A Partitions object
+#' @param named boolean that declares if names should be included
+#' @export
+setMethod(toJSONGeneric, "Partitions", function(object, named = c(TRUE, FALSE)) {
+    named <- veupathUtils::matchArg(named)
+    tmp <- veupathUtils::S4SimpleListToJSON(object, FALSE)
+
+    if (named) tmp <- paste0('{"partitions":', tmp, "}")
+
+    return(tmp)
+})
+
+## TODO reduce repeated code, maybe call the Network method and figure how to add partitions? or a refactor? not urgent.
+#' Convert KPartiteNetwork object to JSON
+#' 
+#' Converts a KPartiteNetwork object to JSON
+#' @param object A KPartiteNetwork object
+#' @param named boolean that declares if names should be included
+#' @export 
+setMethod(toJSONGeneric, "KPartiteNetwork", function(object, named = c(TRUE, FALSE)) {
+    named <- veupathUtils::matchArg(named)
+    tmp <- character()
+
+    nodes_json <- veupathUtils::toJSON(object@nodes, named = FALSE)
+    links_json <- veupathUtils::toJSON(object@links, named = FALSE)
+    partitions_json <- veupathUtils::toJSON(object@partitions, named = FALSE)
+    
+    # TODO this doesnt conform to the api in the data service, bc there we explicitly have a bipartite network and not a kpartite
+    # we have `columns1NodeIds` and `columns2NodeIds` instead of `partitions`. i think this is better though.
```

Comment:
to be clear, are you suggesting an api change? If yes, when do you think we could have this wired and tested? It's possible we could get this out with 67

Reply:
yes, im suggesting an api change. the work for that would be in the data service and monorepo. and i think if we do it, it should be for b67